File: native/SetupWizard.swift
import SwiftUI

struct SetupWizard: View {
    let runtime: URL
    let pairingAllowed: Bool
    let finish: (Bool) -> Void
    @State private var step = 0
    @State private var sources = Dictionary(uniqueKeysWithValues: CollectorConfiguration.defaults.keys.map { ($0, false) })
    @State private var failure = ""
    private let titles = ["Welcome to Observatory", "Choose what to collect", "Connect your devices"]
    var body: some View {
        VStack(alignment: .leading, spacing: 20) {
            Text("Step \(step + 1) of 3").font(.caption).foregroundStyle(.secondary)
            Text(titles[step]).font(.largeTitle.bold())
            if step == 0 {
                Text("Track screen time and AI usage across your devices. You choose which sources are enabled. Collection stays off until you finish setup.")
                Text("Dashboard snapshots exclude prompts, window titles, transcripts, audio and credentials. Provider sign-ins stay on their owning device.")
                Text("This release uses direct encrypted device pairing. There is no Observatory account or hosted sync service.")
            } else if step == 1 {
                ForEach([("activity", "ActivityWatch screen time"), ("codex", "Saved Codex usage and settings"),
                         ("quota", "Codex account limits online"), ("wispr", "Wispr Flow statistics")], id: \.0) { key, title in
                    Toggle(title, isOn: Binding(get: { sources[key] == true }, set: { sources[key] = $0 }))
                }
                Text("ActivityWatch must already be running. Account monitoring uses the installed Codex sign-in and reads online limits without making model requests. Other sources need their applications and local records. Enabling a source does not prove it is available.")
                    .font(.callout).foregroundStyle(.secondary).fixedSize(horizontal: false, vertical: true)
            } else {
                Text("Pair with Windows to share supported sanitized usage records directly. Provider credentials are never transferred. Account-limit history is not synchronized yet.")
                Text("The current connection requires an existing SSH alias, trusted host key and key-based sign-in. The pairing dialog shows the destination and scope before sending setup. Automatic discovery is not available yet.")
                Text("You can finish on this Mac now and pair later. Canceling pairing does not mark a device connected.")
            }
            if !failure.isEmpty { Text(failure).foregroundStyle(.red) }
            Spacer()
            HStack {
                if step > 0 { Button("Back") { step -= 1 } }
                Spacer()
                if step < 2 {
                    Button("Continue") { step += 1 }.keyboardShortcut(.defaultAction)
                } else {
                    Button("Finish on this Mac") { complete(pair: false) }
                    Button("Finish and pair Windows…") { complete(pair: true) }.disabled(!pairingAllowed)
                }
            }
        }.padding(30).frame(width: 600, height: 500)
    }
    private func complete(pair: Bool) {
        do {
            try FirstRunSetup.complete(sources: sources, runtime: runtime)
            finish(pair)
        } catch { failure = "Setup could not be saved. Collection remains paused. Try again after checking the local configuration." }
    }
}

File: native/windows/SetupWizard.cs
namespace WorkspaceObservatory;

internal sealed class SetupWizard : Form
{
    private readonly Collector collector;
    private readonly string runtime;
    private readonly Dictionary<string, bool> selected = new() { ["activity"] = false, ["codex"] = false, ["wispr"] = false, ["typewhisper"] = false, ["quota"] = false, ["wsl"] = false };
    private readonly FlowLayoutPanel content = new() { Dock = DockStyle.Fill, FlowDirection = FlowDirection.TopDown, WrapContents = false, AutoScroll = true, Padding = new Padding(24) };
    private readonly Button next = new() { Text = "Continue", AutoSize = true };
    private readonly Button back = new() { Text = "Back", AutoSize = true };
    private int step;
    private bool ubuntuAccount;

    internal SetupWizard(string runtime, Collector collector)
    {
        this.runtime = runtime; this.collector = collector;
        Text = "Set up Observatory"; ClientSize = new Size(620, 520);
        Font = new Font("Segoe UI", 10); StartPosition = FormStartPosition.CenterScreen;
        FormBorderStyle = FormBorderStyle.FixedDialog; MaximizeBox = false; MinimizeBox = false;
        var footer = new FlowLayoutPanel { Dock = DockStyle.Bottom, Height = 58, FlowDirection = FlowDirection.RightToLeft, Padding = new Padding(12) };
        footer.Controls.Add(next); footer.Controls.Add(back); Controls.Add(content); Controls.Add(footer);
        next.Click += (_, _) => Advance(); back.Click += (_, _) => { step--; Render(); };
        AcceptButton = next; Render();
    }
    private void Paragraph(string text, bool heading = false)
    {
        content.Controls.Add(new Label { Text = text, AutoSize = true, MaximumSize = new Size(550, 0), Margin = new Padding(0, 0, 0, 16),
            Font = heading ? new Font(Font.FontFamily, 20, FontStyle.Bold) : Font });
    }
    private void Render()
    {
        foreach (var control in content.Controls.Cast<Control>().ToArray()) control.Dispose();
        back.Visible = step > 0; next.Text = step == 2 ? "Finish on this PC" : "Continue";
        Paragraph($"Step {step + 1} of 3");
        Paragraph(new[] { "Welcome to Observatory", "Choose what to collect", "Connect your devices" }[step], true);
        if (step == 0)
        {
            Paragraph("Collection stays off until you finish setup. Choose the sources you want to read on this PC.");
            Paragraph("Dashboard snapshots exclude prompts, window titles, transcripts, audio and credentials. Provider sign-ins stay on their owning device.");
            Paragraph("This release uses direct encrypted device pairing, without an Observatory account or hosted sync service.");
        }
        else if (step == 1)
        {
            foreach (var (key, title) in new[] { ("activity", "ActivityWatch screen time"), ("codex", "Saved Codex usage and settings"), ("wispr", "Wispr Flow statistics"), ("typewhisper", "TypeWhisper aggregate statistics"), ("wsl", "Include Ubuntu WSL records with saved Codex"), ("quota", "Codex account limits online") })
            {
                var check = new CheckBox { Text = title, Tag = key, Checked = selected[key], AutoSize = true, Margin = new Padding(0, 0, 0, 8), Enabled = key != "wsl" || selected["codex"] };
                check.CheckedChanged += (_, _) =>
                {
                    selected[key] = check.Checked;
                    if (key == "codex")
                    {
                        var wsl = content.Controls.OfType<CheckBox>().FirstOrDefault(item => Equals(item.Tag, "wsl"));
                        if (wsl is not null) { wsl.Enabled = check.Checked; if (!check.Checked) wsl.Checked = false; }
                    }
                    if (key == "quota")
                    {
                        var source = content.Controls.OfType<CheckBox>().FirstOrDefault(item => Equals(item.Tag, "account"));
                        if (source is not null) source.Enabled = check.Checked;
                    }
                };
                content.Controls.Add(check);
            }
            var account = new CheckBox { Text = "Use Ubuntu's Codex account instead of Windows", Tag = "account", Checked = ubuntuAccount, AutoSize = true, Enabled = selected["quota"] };
            account.CheckedChanged += (_, _) => ubuntuAccount = account.Checked;
            content.Controls.Add(account);
            Paragraph("Sources need their installed applications and records. Ubuntu options may start WSL. Online monitoring uses that selected client's existing sign-in, with no credential copying or fallback to another account. Enabling a source does not prove it is available.");
        }
        else
        {
            Paragraph("Pair from the Mac using this PC's pairing details, available in the system-tray menu after setup. Review the destination and shared-data scope before confirming.");
            Paragraph("The current connection requires an existing SSH alias, trusted host key and key-based sign-in. Automatic discovery is not available yet. Account-limit history is not synchronized.");
            Paragraph("Finish here to use this PC alone or pair later. Finishing setup does not claim a device is connected.");
        }
    }
    private void Advance()
    {
        if (step < 2) { step++; Render(); return; }
        try
        {
            FirstRunSetup.Complete(runtime, () => collector.Configure(selected["wsl"] ? "Ubuntu" : null, selected["wispr"], selected["quota"],
                selected["quota"] && ubuntuAccount ? "Ubuntu" : null, selected["activity"], selected["codex"], selected["typewhisper"]));
            DialogResult = DialogResult.OK; Close();
        }
        catch { MessageBox.Show("Setup could not be saved. Collection remains paused. Check local storage and try again.", "Setup incomplete"); }
    }
}
